docs/tutorial/03-adding-in-transactions.md

```diff
@@ -0,0 +1,508 @@
+# Adding in Transactions
+
+Now comes the exciting part: adding some economic incentives to our guessing game! We'll implement a system where users pay to play and winners take the entire pot. This is where blockchain development gets really interesting. And honestly, it's probably why you're here in the first place, right?
+
+## What We'll Accomplish
+
+By the end of this step, you'll have:
+
+- A guessing fee that players must pay
+- A prize pot that accumulates all the fees
+- Automatic winner payouts
+- Admin funding of the initial prize pot
+- Understanding of token transfers in smart contracts
+
+## Understanding the Economic Model
+
+Here's how our game economics will work:
+
+1. **Admin funds the pot**: When resetting, admin transfers XLM to the contract
+2. **Players pay to play**: Each guess costs a small fee (added to the pot)
+3. **Winner takes all**: Correct guesses win the entire accumulated pot
+4. **New round starts**: Admin can reset with fresh funding
+
+This creates real stakes and makes the game much more engaging!
+
+## Step 1: 🪙 Add Token Handling
+
+First, we need to import Stellar's token functionality. Add this to your imports at the top of `lib.rs`:
+
+```rust
+#![no_std]
+use admin_sep::{Administratable, Upgradable};
+use soroban_sdk::{
+    contract, contractimpl, symbol_short, token, Address, Env, Symbol
+};
+```
+
+The `token` import gives us access to Stellar's built-in token functionality.
+
+## Step 2: 💰 Add Economic Storage Variables
+
+Let's add storage for tracking our economic state. Add these constants:
+
+```rust
+const THE_NUMBER: Symbol = symbol_short!("n");
+const LAST_GUESSER: Symbol = symbol_short!("guesser");
+const PRIZE_POT: Symbol = symbol_short!("pot");        // Amount in the prize pot
+const GUESS_FEE: Symbol = symbol_short!("fee");        // Cost per guess
+const NATIVE_TOKEN: Symbol = symbol_short!("native");   // XLM token address
+```
+
+## Step 3: Update the Constructor
+
+Let's enhance our constructor to set up the economic parameters:
+
+```rust
+pub fn __constructor(env: &Env, admin: &Address, initial_pot: u64, guess_fee: u64) {
```

Comment:
I think you mean to remove this initial_pot argument, since later you say the pot starts empty